### PR TITLE
fix: headers rewrite example configuration

### LINKED ISSUE #35816: Incorrect `teleport.dev/app-rewrite` annotation example
## Applies To

https://goteleport.com/docs/application-access/enroll-kubernetes-applications/reference/

## Details

Current example of teleport.dev/app-rewrite annotation for App discovery set up headers rewriting incorrectly and cause errors in agent log and skipping the App to be added, 

## How will we know this is resolved?

Annotation headers rewrite example should use this way:
```          
          teleport.dev/app-rewrite: |
            headers:
            - name: "Host"
              value: "grafana.teleport.k8s.olegen.fun"
            - name: "Origin"
              value: "https://grafana.teleport.k8s.olegen.fun" 
```

### PR DESCRIPTION
**Pull request by olegenii**
Closes #35816

Fix headers rewrite example configuration in teleport.dev/app-rewrite annotation for App discovery

### CODE REVIEW

**Review by mdwn — APPROVED** [review]

**Comment by ptgott** [comment]
@olegenii Thanks for your contribution! The next step will be for us to open an internal branch, copy your commits onto it, and open a buddy PR. You can expect this shortly.

File: docs/pages/application-access/enroll-kubernetes-applications/reference.mdx [hunk 1]
```diff
@@ -108,7 +108,9 @@ annotations:
     - "localhost"
     - "jenkins.internal.dev"
     headers:
-    - "X-Custom-Header: example"
-    - "Authorization: Bearer {{internal.jwt}}"
+    - name: "X-Custom-Header"
```
Comment (zmb3) [inline docs/pages/application-access/enroll-kubernetes-applications/reference.mdx:111]: This no longer matches the link above, so the _same as one would put into rewrite config section_ text is not true and needs to be updated.

Comment (olegenii) [reply docs/pages/application-access/enroll-kubernetes-applications/reference.mdx:111]: > This no longer matches the link above, so the _same as one would put into rewrite config section_ text is not true and needs to be updated.

[Dynamic registration syntax](https://goteleport.com/docs/application-access/guides/connecting-apps/#rewrite-redirect) tab already has the right syntax for header rewriting. Is that what you mean?

Comment (zmb3) [reply docs/pages/application-access/enroll-kubernetes-applications/reference.mdx:111]: Yeah, but the dynamic registration syntax isn't selected by default, so if someone follows the link here and takes what they see on screen and dumps it into the `app-rewrite` annotation, they're likely going to make a mistake.

Maybe we just need to say something like _same as one would use when configuring an app with dynamic registration._

Comment (olegenii) [reply docs/pages/application-access/enroll-kubernetes-applications/reference.mdx:110]: > Yeah, but the dynamic registration syntax isn't selected by default, so if someone follows the link here and takes what they see on screen and dumps it into the `app-rewrite` annotation, they're likely going to make a mistake.
> 
> Maybe we just need to say something like _same as one would use when configuring an app with dynamic registration._

Added

**Review by zmb3 — APPROVED** [review]

**Pull request closed without merging**